source/glkernel/include/glkernel/sample.hpp

```diff
@@ -207,6 +208,33 @@ size_t poisson_square(tkernel<glm::tvec2<T, P>> & kernel, const T min_dist, cons
     return k + 1;
 }
 
+template <typename T, glm::precision P>
+size_t n_rooks(tkernel<glm::tvec2<T, P>> & kernel)
+{
+	assert(kernel.depth() == 1);
+	assert(kernel.width() == kernel.height());
+
+	auto stratum_size = 1.0 / kernel.width();
+	std::random_device RD;
+	std::mt19937_64 generator(RD());
+	std::uniform_real_distribution<> jitter_dist(0.0, stratum_size);
+
+	std::vector<unsigned int> pool;
+	for (unsigned int k = 0; k < kernel.width(); ++k)
```

Comment:
we need to think about the kernel semantics here: hmm, as far as i understand the n-rooks sampling, the number of samples should be ```kernel.size()```. It should work for any kernel dimension (and works for 2, 3, as well as for 4 coefficients ?, but we can ignore this for now) ... 

I would expect the function to fill all values within my kernel - having a 2D kernel of arbitrary width and height, it should result in width * height samples that use #size strata vertically and #size strata horizontally ... ?